docs/en/latest/practices/how-to-use-go-plugin-runner-in-apisix-ingress.md

```diff
@@ -0,0 +1,207 @@
+
+#### Load the image to the cluster environment
+
+```bash
+kind load docker-image apisix/forrunner:0.1
+```
+
+Load the image into the kind cluster environment to pull the custom local image for installation during the helm installation.
+
+### Install APISIX Ingress
+
+#### Create namespace
+
+Then install APISIX using helm with the following command in the directory of Apache APISIX Helm Chart:
+
+```bash
+helm install apisix apisix/apisix --set gateway.type=NodePort --set apisix.image.repository=custom/apisix --set apisix.image.tag=v0.1 --set extPlugin.enabled=true --set extPlugin.cmd=["/usr/local/apisix-go-plugin-runner/go-runner", "run"] --set ingress-controller.enabled=true --set ingress-controller.config.apisix.serviceNamespace=apisix --namespace apisix --create-namespace --set ingress-controller.config.apisix.serviceName=apisix-admin
+```
+
+### Create httpbin service and ApisixRoute resources
+
+Create an httpbin backend resource to run with the deployed ApisixRoute resource to test that the functionality is working correctly.
+
+#### Create httpbin service
+
+Create an httpbin service with the following command:
+
+```bash
+kubectl run httpbin --image kennethreitz/httpbin --port 80
+```
+
+Expose the port with the following command:
+
+```bash
+kubectl expose pod httpbin --port 80
+```
+
+#### Create ApisixRoute Resource
+
+Create the `go-plugin-runner-route.yaml` file to enable the ApisixRoute resource, with the following configuration file:
+
+```yaml
+apiVersion: apisix.apache.org/v2beta3
+kind: ApisixRoute
+metadata:
+  name: plugin-runner-demo
+spec:
+  http:
+  - name: rule1
+    match:
+      hosts:
+      - local.httpbin.org
+      paths:
+      - /get
+    backends:
+    - serviceName: httpbin
+      servicePort: 80
+    plugins:
+    - name: ext-plugin-pre-req
+      enable: true
+      config:
+        conf:
+        - name: "say"
+          value: "{\"body\": \"hello\"}"
+```
+
+The create resource command is as follows:
+
+```bash
+kubectl apply -f go-plugin-runner-route.yaml
+```
+
+### Test
+
+The command is as follows to test if the plugin written in Golang is working correctly:
+
+```bash
+kubectl exec -it -n ${namespace of Apache APISIX} ${Pod name of Apache APISIX} -- curl http://127.0.0.1:9080/get -H 'Host: local.httpbin.org'
+```
+
+Here I derived from the `kubectl get pods --all-namespaces` command that the `${namespace of Apache APISIX}` and `${Pod name of Apache APISIX}` parameters here are `ingress-apisix` and `apisix- 55d476c64-s5lzw`, execute the command as follows:
+
+```bash
+kubectl exec -it -n ingress-apisix apisix-55d476c64-s5lzw -- curl http://127.0.0.1:9080/get -H 'Host: local.httpbin.org'
+```
+
+The expected response obtained is:
+
+```bash
+chever@cloud-native-01:~/api7/cloud_native/tasks/plugin-runner$ kubectl exec -it -n ingress-apisix apisix-55d476c64-s5lzw -- curl http://127.0.0.1:9080/get -H 'Host: local.httpbin.org'
```

Comment:
```suggestion
kubectl exec -it -n ingress-apisix apisix-55d476c64-s5lzw -- curl http://127.0.0.1:9080/get -H 'Host: local.httpbin.org'
```

The format here can be consistent with the previous format

Reply:
Solved